network/common/src/main/java/org/apache/spark/network/buffer/LargeByteBuffer.java

```diff
@@ -0,0 +1,113 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.spark.network.buffer;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+public interface LargeByteBuffer {
```

Comment:
Interfaces / classes should have Javadoc to explain them.  For this interface in particular, a casual reader of the code might wonder why we wouldn't just use Java's `Buffer` or `ByteBuffer` interfaces.  I think that the reason is that we need `position`, `size` / `limit`, etc. to return longs instead of integers in order to support larger buffer sizes.  It would be good to clarify this.